interactions/four_particle/__init__.py

```diff
@@ -137,6 +139,14 @@ def integrate(self, ps, stepsize=None):
         # All matrix elements share the same weak scale multiplier
         unit = 32 * CONST.G_F**2
         constant = unit * (params.aT / params.a)**5 / 64. / numpy.pi**3 / params.H
+
+        left = Counter(item.specie for item in self.reaction if item.side == -1)
+        right = Counter(item.specie for item in self.reaction if item.side == 1)
+        if left[self.reaction[0].specie] == 2 and right[self.reaction[0].specie] in [0, 1]:
+            constant *= 2
+        if left[self.reaction[0].specie] == 3 and right[self.reaction[0].specie] == 0:
+            constant *= 3
+
```

Comment:
This looks *extremely* suspicious. Why do you double the collision integral for a neutrino with momentum `p0` if there was created a neutrino with momentum `p2`?

Reply:
This is the same as adding a copy of the same collision integral for the other neutrino? We still calculate it for every momentum on the grid.

Reply:
So the thing for which you are compensating will be computed elsewhere. You are double-counting.

Reply:
Have you checked the SM decoupling spectra? I bet they're wrong

Reply:
Where will it else be computed? 
In the equation

dn_comoving / dt = I_coll

for the reaction N --> 3 x nu

we will need n_comoving += 3 * dn_comoving , right?

Where does the code take this into account?

Reply:
The code takes this into account when one decay event contributes to I_coll(nu) at 3 places corresponding to momenta p2, p3 and p4. So when you integrate the distribution function, you count everything properly